bids/layout/layout.py

```diff
@@ -753,6 +753,9 @@ def _build_file_query(self, **kwargs):
                 if val is None or val == Query.NONE:
                     join_method = query.outerjoin
                     val_clause = tag_alias._value.is_(None)
+                elif val == Query.ANY_OPTIONAL:
+                    join_method = query.outerjoin
+                    val_clause = sa.or_(tag_alias._value.is_(None), tag_alias._value.isnot(None))
```

Comment:
Since we're replicating the behavior of not including the filter at all:

```suggestion
                elif val == Query.ANY_OPTIONAL:
                    continue
```